### PR TITLE
♻️Separate build attributes out into a new class

### PR DESCRIPTION
**Pull request by rosiemiller2**
### Description
Created an `MbedTargetBuildAttributes` class and methods for accessing this with either a board type or an `MbedTarget` instance.

This class doesn't yet have all the fields needed by mbed-build, but I would rather we added them as we need them. That way we can keep track of what parts of the targets.json definition are actually being used and what are just relics of older configuration.

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_targets/mbed_target_build_attributes.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        supported_toolchains: Toolchains that can be used to build for this target.
+        default_toolchain: Default toolchain used for this target.
+    """
+
+    labels: FrozenSet[str]
+    features: FrozenSet[str]
+    components: FrozenSet[str]
+    config: Dict[str, str]
+    supported_toolchains: FrozenSet[str]
+    default_toolchain: str
```
Comment (muchzill4) [inline mbed_targets/mbed_target_build_attributes.py]: This should be `Optional[str]` as below `get('', None)` is used.

File: mbed_targets/get_build_attributes.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    The build attributes are as defined in the targets.json file linked to.
+
+    Args:
+        mbed_target: the Mbed Target object representing the target data from the online database
+        path_to_targets_json: path to Mbed OS's targets.json file
+
+    Raises:
+        TargetBuildAttributesError: an error has occurred while fetching build attributes
+    """
+    return MbedTargetBuildAttributes.from_board_type(mbed_target.board_type, path_to_targets_json)
```
Comment (muchzill4) [inline mbed_targets/get_build_attributes.py]:
```suggestion
    return get_build_attributes_by_board_type(mbed_target.board_type, path_to_targets_json)
```

Thoughts? **It'd need updating the test, so don't commit!**

File: mbed_targets/mbed_target_build_attributes.py (hunk 1 above)
Comment (muchzill4) [reply mbed_targets/mbed_target_build_attributes.py]: Shows how good `mypy` is, eh? ;)

Comment (rosiemiller2) [reply mbed_targets/mbed_target_build_attributes.py]: efc3f58

File: mbed_targets/get_build_attributes.py (hunk 2 above)
Comment (rosiemiller2) [reply mbed_targets/get_build_attributes.py]: Yeah I like it 82cfc04.

**Pull request merged**